src/parser.rs

```diff
@@ -1,35 +1,63 @@
 use html5ever::{self, Attribute};
 use html5ever::tree_builder::{TreeSink, NodeOrText, QuirksMode};
 use std::borrow::Cow;
+use std::fs::File;
+use std::path::Path;
+use std::io::Read;
 use string_cache::QualName;
 use typed_arena::Arena;
 
+
 use tree::Node;
+use std::option;
+
+pub struct Html<F = IgnoreParseErrors> where F: FnMut(Cow<'static, str>) {
+    opts: ParseOpts<F>,
+    data: option::IntoIter<String>,
+}
 
+impl Html  {
+    pub fn from_string(string: &str) -> Html {
```

Comment:
Fixed, but while we are on the subject of input, is there a way to support `Into<Iterator<Item=u8>>`? 

Ilogiq on [Reddit](http://www.reddit.com/r/rust/comments/3878qz/feedback_gatheringhelp_me_design_api_for_kuchiki/) asked for such interface, although I'm unsure if you can transform such input into `Into<String>` without allocation.